### PR TITLE
Fixes for performance

### PR DESCRIPTION
**Pull request by iopq**
So far I couldn't get any more improvements past this

### CODE REVIEW

**Comment by ujh** [comment]
I ran the benchmark on my machine, too. And I couldn't see any speed improvements at all. So I suggest we don't merge this right now as the compile time goes up dramatically and slows down development. At least it slows we down. :)

**Comment by ujh** [comment]
BTW, I only closed this pull request due to the inlining. If there's other things (you were talking about a bug fix yesterday) feel free to open separate pull requests for those bits.

**Comment by iopq** [comment]
I took out some of the inlining. I can enable it back by doing a search on the source code and replacing it before a build anyway.

**Comment by iopq** [comment]
This gives me 15% higher throughput on playouts

File: src/board/chain/test.rs [hunk 1]
```diff
@@ -3,10 +3,13 @@
 use board::Black;
 use board::Chain;
 use board::Coord;
+use std::collections::HashSet;
 
 #[test]
 fn show_returns_a_legible_string_for_the_chain() {
-    let mut c = Chain::new(1, Black, Coord::new(7,7), vec!(Coord::new(1,1)));
+	let mut set = HashSet::new();
```
Comment (ujh) [inline src/board/chain/test.rs]: Four spaces for indentation please.

File: src/board/mod.rs [hunk 2]
```diff
@@ -32,14 +32,14 @@ use ruleset::Ruleset;
 use score::Score;
 use self::point::Point;
```
Comment (ujh) [inline src/board/mod.rs]: My OCD is flaring up when I see that. ;) All use statements together and in alphabetical order, please. Pretty please?

File: src/board/mod.rs [hunk 3]
```diff
@@ -243,19 +257,90 @@ impl Board {
...
+	}
+    
+    pub fn playout_moves(&self) -> Vec<Move> {
+        	let color = self.next_player();
+            self.vacant
+                .iter()
+                .map(|coord| Play(color, coord.col, coord.row))
+        	    .filter(|m| !self.is_eye(&m.coord(), *m.color()) && self.playout_legal_move(*m) )
+                .collect()
+	}
```
Comment (ujh) [inline src/board/mod.rs]: Doesn't this duplicate the code from `legal_moves`? If so, why are we doing this?

File: src/board/mod.rs [hunk 4]
```diff
@@ -243,19 +257,90 @@ impl Board {
             let color = self.next_player();
             self.vacant
                 .iter()
-                .map(|coord| Play(color.clone(), coord.col, coord.row))
+                .map(|coord| Play(color, coord.col, coord.row))
                 .filter(|m| self.is_legal(*m).is_ok())
                 .collect()
         }
     }
-
```
Comment (ujh) [inline src/board/mod.rs]: The idea here is to pick a random legal move and the use it if it satisfies the eye check, right? I'd prefer to have this code inside of playout. Just put it there, I'm in the process of writing tests for that part anyway.

File: src/board/chain/test.rs (hunk 1 above)
Comment (iopq) [inline src/board/chain/test.rs]: I don't know how that's even possible, eclipse is supposed to replace tab with spaces

File: src/board/mod.rs (hunk 3 above)
Comment (iopq) [inline src/board/mod.rs]: it was faster, but then I found an even faster approach

File: src/board/mod.rs (hunk 4 above)
Comment (iopq) [inline src/board/mod.rs]: I tried to put the code directly into the function, but I got a borrowing error about borrowing board. I just put the code into board directly and for some reason it worked. The code definitely needs cleanup, though.

File: src/board/mod.rs (hunk 2 above)
Comment (iopq) [inline src/board/mod.rs]: But it IS in alphabetical order.

Comment (ujh) [inline src/board/mod.rs]: Yes, sorry. No coffee and commenting on an iPhone during a commute are my excuses. ;)

Comment (ujh) [inline src/board/mod.rs]: Actually, hold on. What I meant is that I like to keep the setup of two blocks of uses. The first one (alphabetically ordered :stuck_out_tongue_winking_eye:) for stuff from our own repo. And then a second block with the stuff from the stdlib and external crates. So the `rand` stuff needs to go down and `std::sync::Arc` needs to go to the spot where it was (ordered by full name, please).

File: src/board/mod.rs (hunk 4 above)
Comment (ujh) [inline src/board/mod.rs]: OK, so let's leave it there for now. I'll have a look at it in my branch as I work on having two kinds of playouts (one with the eye check and one without it) anyway.

**Comment by ujh** [comment]
And sorry about all those coveralls comments! That's really annoying and I've now turned that off.

**Comment by ujh** [comment]
Whoa! I just checked the results. Here's what I got on my machine:

* 4794pps on 9x9 (1.87x faster than before)
* 2342pps on 13x13 (2x4x faster than before)
* 988pps on 19x19 (3.5x faster than before)

That's quite nice already!

**Pull request merged**